winterfell/webapp/src/js/profileService.js

```diff
@@ -1,12 +1,47 @@
 'use strict';
 var app = angular.module('vetafiApp');
-app.factory('profileService', ['claimService', function(claimService) {
-  return {
-    userInfo: {},
-    clearUserInfo: function() {
-      sessionStorageHelper.removePair(vfiConstants.keyUserId);
-      this.userInfo = {};
+app.service('Profile', ['claimService', '$q', 'net', function(claimService, $q, net) {
+    this.user = null;
+
+    this.logout = function() {
       claimService.clearClaims();
-    }
-  };
+      this.unsetUser();
+    };
+
+    this.setUser = function(user) {
+      this.user = user;
+    };
+
+    this.unsetUser = function() {
+      this.user = null;
+    };
+
+    this.isSetUser = function() {
```

Comment:
I feel like `has` is for collection-like objects, `isSet` to me makes more sense for instance - fields